### ISSUE TITLE
[Bug] Uniqueness test on company_id in company history model

### ISSUE THREAD

**Issue by matthew-LL** [fivetran/dbt_intercom_source#30]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

The `stg_intercom__company_history` model has a uniqueness test on `company_id`, however I have a encountered an error since `company_id` is not unique on the history table as a new row per company is added whenever the company is updated. This results in a `company_id` uniqueness test failure when building `stg_intercom__company_history`.

```
  - name: stg_intercom__company_history
    description: Table containing historical information about companies which have been created within Intercom.
    columns:
      - name: company_id
        description: The unique identifier representing the Intercom defined company.
        tests:
          - not_null
          - unique
```

### Expected behavior

I would expect primary key tests for `stg_intercom__company_history` to pass only if they were testing a surrogate key based off `company_id` and `updated_at`.

### dbt Project configurations

```  
intercom_database: raw_db
intercom_schema: fivetran_intercom
intercom__using_company_tags: false
```

### Package versions

```
  - package: fivetran/intercom_source
    version: [">=0.7.0", "<0.8.0"]
```

### What database are you using dbt with?

snowflake

### dbt Version

```
Core:
  - installed: 1.5.6
```

### Are you willing to open a PR to help address this issue?

- [X] Yes.
- [ ] Yes, but I will need assistance and will schedule time during our [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours) for guidance
- [ ] No.

**Comment by fivetran-joemarkiewicz** [comment]
Hi @matthew-LL thanks for opening this issue and providing context as to the test failure you are seeing.

I completely agree with you that the below uniqueness test should not be included in the staging model.

https://github.com/fivetran/dbt_intercom_source/blob/aa55a0db1e9b08236197173bb74b531ae96c40e8/models/stg_intercom.yml#L19-L26

It actually looks like for the other history staging models do not have uniqueness tests and instead are testing on the ids not being null. As such, I would proposed for this immediate bugfix we simply remove the uniqueness test. We can then open a FR to overhaul the testing to be more robust and test on the surrogate key for all history tables to be that of the id and updates_at fields.

Would you be interested in opening a PR to address the removal of the uniqueness test for this immediate fix?

**Comment by matthew-LL** [comment]
Hi @fivetran-joemarkiewicz , thanks for getting back to me on this so quickly, I'd be happy to open a PR to remove this test!

**Comment by fivetran-joemarkiewicz** [comment]
That sounds great! Let me know if you need any help or guidance when creating the PR 😄

**Comment by fivetran-joemarkiewicz** [comment]
Thanks for opening the PR @matthew-LL! ❤️ 

I will review this in the current week and get back to you with any comments if I have any.

**Comment by fivetran-catfritz** [comment]
Closing this issue as this change was released on v0.7.1. Added feature request #32 to revise the testing used on staging history tables.
